### ISSUE TITLE
SYSLIB0050: Formatter-based serialization is obsolete

### ISSUE THREAD

**Issue by atlemann** [stephan-tolksdorf/fparsec#110]
NET8.0 has marked a bunch of serializable interfaces as obsolete, which will be removed in NET9.0. I guess this means this library won't work anymore when clients eventually upgrade their apps NET9.0?

It seems to be mostly an issue with the non low trust version and the use of IObjectReference interface and other things from  FormatterServices

The `System.Runtime.Serialization.FormatterServices.GetUninitializedObject` method can be replaced with `System.Runtime.CompilerServices.RuntimeHelpers.GetUninitializedObject`, but not sure about the custom Cloning and CharStream classes.

And those classes are doing some dark magic and don't seem trivial to replace.

**Comment by stephan-tolksdorf** [comment]
Yes, the Cloning and StringBuffer stuff has to be replaced in the non-low-trust version. 
I have draft code for that, but don't yet know how quickly I will get that into something ready for a PR.

**Comment by atlemann** [comment]
It would be totally awesome if you have something. But I can only imagine the amount of work involved in replacing that code.

**Comment by ingted** [comment]
Any progress on this?

**Comment by Martin521** [comment]
Try [XParsec](https://github.com/roboz0r/XParsec)
